samples/HttpClientFactorySample/Program.cs

```diff
@@ -19,19 +19,23 @@ public class Program
         public static async Task Run()
         {
             var services = new ServiceCollection()
-                .AddHttpClient()
+                .AddHttpClient("github", c =>
+                {
+                    c.BaseAddress = new Uri("https://api.github.com/");
```

Comment:
This is nice because it's the existing API. We don't have to worry about keeping things in sync. What does consumption look like? Do you do a using around the created http client since we're going to be caching handlers?

Reply:
>This is nice because it's the existing API.

Yeah, I'm happy about how this worked out.

> What does consumption look like? 

Not sure what you mean. What's there is there in the sample. We've had some offline discussions about usage in MVC already but that's not part of this work-item.

> Do you do a using around the created http client since we're going to be caching handlers?

You can, you don't really have to. The only thing the using would do is cancel any requests that are going on.